src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -29,6 +32,152 @@
 }
 
 #if defined(TARGET_MACCATALYST) || defined(TARGET_IOS) || defined(TARGET_TVOS)
+
+const char* NativeGetLocaleName(const char* localeName,
+                                         int32_t valueLength)
+{
+     NSString *locName = [NSString stringWithFormat:@"%s", localeName];
+     NSLocale *currentLocale = [[NSLocale alloc] initWithLocaleIdentifier:locName];
+     const char* value = [currentLocale.localeIdentifier UTF8String];
+     return strdup(value);
+}
+
+const char* NativeGetLocaleInfoString(const char* localeName,
+                                                LocaleStringData localeStringData,
+                                                int32_t valueLength,
+                                                const char* uiLocaleName)
+{
+    const char* value;
+    NSString *locName = [NSString stringWithFormat:@"%s", localeName];
+    NSLocale *currentLocale = [[NSLocale alloc] initWithLocaleIdentifier:locName];
+    NSNumberFormatter *numberFormatter = [[NSNumberFormatter alloc] init];
+    numberFormatter.locale = currentLocale;
+    NSDateFormatter* dateFormatter = [[NSDateFormatter alloc] init];
+    [dateFormatter setLocale:currentLocale];
+    NSLocale *gbLocale = [[NSLocale alloc] initWithLocaleIdentifier:@"en_GB"];
+
+
+    switch (localeStringData)
```

Comment:
There seems to be a good chunk of TODOs/uncertainty of correctness, should we group them all at the end and throw a not implemented error or something until we are sure? And put them all in one tracking issue?

Reply:
yes, that's a good idea. Will update.